src/bundle.ts

```diff
@@ -61,213 +60,40 @@ export class Bundler extends Transform {
 
   _flush(done: (error?: any) => void) {
     this._buildBundles().then((bundles: Map<string, string>) => {
-      for (const fragment of this.config.allFragments) {
-        const file = this.analyzer.getFile(fragment);
-        console.assert(file != null);
-        const contents = bundles.get(fragment);
+      for (const filename of bundles.keys()) {
+        const filepath = pathFromUrl(this.config.root, filename);
+        let file =
+            this.analyzer.getFile(filepath) || new File({path: filepath});
+        const contents = bundles.get(filename);
         file.contents = new Buffer(contents);
         this.push(file);
       }
-      const sharedBundle = bundles.get(this.sharedBundleUrl);
-      if (sharedBundle) {
-        const contents = bundles.get(this.sharedBundleUrl);
-        this.sharedFile.contents = new Buffer(contents);
-        this.push(this.sharedFile);
-      }
       // end the stream
       done();
     });
   }
 
   async _buildBundles(): Promise<Map<string, string>> {
-    const bundles = await this._getBundles();
-    const sharedDepsBundle = (this.config.shell) ?
-        urlFromPath(this.config.root, this.config.shell) :
-        this.sharedBundleUrl;
-    const sharedDeps = bundles.get(sharedDepsBundle) || [];
-    const promises: Promise<{url: string, contents: string}>[] = [];
-
+    let strategy: BundleStrategy;
+    const fragments = Array.from(this.config.allFragments);
     if (this.config.shell) {
-      const shellFile = this.analyzer.getFile(this.config.shell);
-      console.assert(shellFile != null);
-      const newShellContent = this._addSharedImportsToShell(bundles);
-      shellFile.contents = new Buffer(newShellContent);
+      strategy = generateShellMergeStrategy(
+          urlFromPath(this.config.root, this.config.shell), 2);
```

Comment:
Will you add an option to configure this magic number `2`?
I mean, there should be a way to try the benefit of generating more shared bundles.